src/builtin/capitalize.h

```diff
@@ -0,0 +1,90 @@
+/**
+ *  capitalize.h
+ *
+ *  Built-in "|capitaluze:bool" modifier
+ *
+ *  @author Youri Moll <youri.moll@copernica.com>
+ *  @copyright 2019 Copernica BV
+ */
+
+/**
+ *  Namespace
+ */
+namespace SmartTpl { namespace Internal {
+
+/**
+ *  Class definition
+ */
+class CapitalizeModifier : public Modifier
+{
+public:
+    /**
+     *  Destructor
+     */
+    virtual ~CapitalizeModifier() {};
+
+    /**
+     *  Modify a value object
+     *  @param  input
+     *  @param  params      Parameters used for this modification
+     *  @return Value
+     */
+    VariantValue modify(const Value &input, const SmartTpl::Parameters &params) override
+    {
+        // copy the entire string
+        std::string output(input.toString());   
+    
+        // our delimiters, we will capitalize after any of these characters are encountered
+        const std::string delims = " \t\r\n\f\v";
+    
+        // determine if words with digits will be processed
+        bool processDigits = false;
+    
+        // did the user specify that we should also process words with digits
+        if (params.size() == 1) processDigits = params[0].toBoolean();
+        
+        // helper variable to determine if a word contains a digit
+        bool containsDigit = false;
+
+        // We will loop backwards through our input string
+        for (int i = output.size() - 1; i >= 0; --i)
+        {
+            // check if the word we are processing contains a digit
+            if (isdigit(output[i])) containsDigit = true;
+            
+            // if we have found one of our delimiters in the string
+            if (delims.find(output[i]) != std::string::npos)
+            {
```

Comment:
You can save one level of nesting here by using `if (output[i] is not delimiter) continue`